commons/src/main/java/com/simpletools/calculator/commons/operations/PowerOperation.kt

```diff
@@ -6,14 +6,24 @@ import com.simpletools.calculator.commons.operations.base.Operation
 
 class PowerOperation(baseValue: Double, secondValue: Double) : BinaryOperation(baseValue, secondValue), Operation {
 
-    override fun getResult(): Double {
-        var result = Math.pow(secondValue,baseValue)
-        if (java.lang.Double.isInfinite(result) || java.lang.Double.isNaN(result))
```

Comment:
This handling, to return zero instead of inf or NaN, was inappropriate and fundamentally incorrect for a calculator, so it was removed. An infinite or NaN result will return just that. Tests were created to assert these results are returned in their respective cases.